### PR TITLE
add noise and reverberance version for BinauralWSJ0Mix

### PR DESCRIPTION
**Pull request by huangzj421**
Hi there, 
I have created a noise and reverberance version of BinauralWSJ0Mix datasets and trained with convtasnet-parallel structure. Here are the recipes and not conflicted with the clean version of datasets.
Also, I have trained convtasnet-parallel.yaml again and got a better results which I could share you with the Google Driver.
Thanks.

### CODE REVIEW

**Comment by huangzj421** [comment]
Hi @ycemsubakan , do you have time for checking this PR please?

**Comment by ycemsubakan** [comment]
Hi @huangzj421 , sorry for the delay. I will try to do it soon.

**Comment by ycemsubakan** [comment]
@huangzj421 , Could you add in the README the instructions on how to run the model with these noisy conditions? As before, the script should automatically download and create the corresponding dataset.

**Comment by ycemsubakan** [comment]
For `convtasnet-parallel-noisy.yaml` I am getting this error. You can see the command I used also here.

image

**Comment by ycemsubakan** [comment]
And, I am observing the following error for `convtasnet-parallel-reverb.yaml`:

image

**Comment by huangzj421** [comment]
> For `convtasnet-parallel-noisy.yaml` I am getting this error. You can see the command I used also here.
> 
> image

I have corrected the data generation scripts and now these should work fine.

**Comment by huangzj421** [comment]
> @huangzj421 , Could you add in the README the instructions on how to run the model with these noisy conditions? As before, the script should automatically download and create the corresponding dataset. I would like to try myself if everything works correctly.

The usage is the same as the previous PR which you should only set wsj_root and output_root and it will download and generate data into output_root.

**Comment by huangzj421** [comment]
> And, I am observing the following error for `convtasnet-parallel-reverb.yaml`:
> 
> image
> 
> Please note that I tried `convtasnet-parallel.yaml`, and that seems to work fine. Correction: Actually, I am also observing that the is stuck at 80, and it doesn't go down for `convtasnet-parallel.yaml'. In the previous PR, I tried and it the error was going down as expected.. I am not sure what's happening here..

I am sorry that I don't understand the "Correction" part you said. Also the test check failed but I don't know why.

**Comment by ycemsubakan** [comment]
> 

Actually, if the dataset is already created, we should not require `wsj_root` again, right?

**Comment by ycemsubakan** [comment]
> 

What I mean with the 'correction' part is that, when I run the script, (with `convtasnet_parallel.yaml`, the error does not go down. It is stuck at 80.

**Comment by huangzj421** [comment]
> > 
> 
> Actually, if the dataset is already created, we should not require `wsj_root` again, right?

Yes, if the dataset is done and dynamic mixing don't need it in the hparams["base_folder_dm"].

**Comment by huangzj421** [comment]
> 

OK, let me check it.

**Comment by ycemsubakan** [comment]
> 
Yes, but we would also not need `wsj_root` if the dataset is already created.

**Comment by ycemsubakan** [comment]
> > For `convtasnet-parallel-noisy.yaml` I am getting this error. You can see the command I used also here.
> > image
> 
> I have corrected the data generation scripts and now these should work fine.

Ok, can you push your code after this correction in the data generation script?

**Comment by huangzj421** [comment]
> > > For `convtasnet-parallel-noisy.yaml` I am getting this error. You can see the command I used also here.
> > > image
> > 
> > 
> > I have corrected the data generation scripts and now these should work fine.
> 
> Ok, can you push your code after this correction in the data generation script?

Oh, it doesn't need to push cause I just corrected in my [repo](https://github.com/huangzj421/Binaural-WSJ0Mix).

**Comment by ycemsubakan** [comment]
> > > > For `convtasnet-parallel-noisy.yaml` I am getting this error. You can see the command I used also here.
> > > > image
> > > 
> > > 
> > > I have corrected the data generation scripts and now these should work fine.
> > 
> > 
> > Ok, can you push your code after this correction in the data generation script?
> 
> Oh, it doesn't need to push cause I just corrected in my [repo](https://github.com/huangzj421/Binaural-WSJ0Mix).

I am now getting this error: 

image

**Comment by huangzj421** [comment]
> > > > > For `convtasnet-parallel-noisy.yaml` I am getting this error. You can see the command I used also here.
> > > > > image
> > > > 
> > > > 
> > > > I have corrected the data generation scripts and now these should work fine.
> > > 
> > > 
> > > Ok, can you push your code after this correction in the data generation script?
> > 
> > 
> > Oh, it doesn't need to push cause I just corrected in my [repo](https://github.com/huangzj421/Binaural-WSJ0Mix).
> 
> I am now getting this error:
> 
> image
It seems that the download [link](https://data.deepai.org/DemandDataset.zip) is invalid. I am trying to search another one.

**Comment by huangzj421** [comment]
> > 
> 
> What I mean with the 'correction' part is that, when I run the script, (with `convtasnet_parallel.yaml`, the error does not go down. It is stuck at 80.

What is the unit of 80? 80%?

**Comment by huangzj421** [comment]
Hi @ycemsubakan , I have fixes some errors in train.py and the data generation scripts in my repo. After you remove the previous BinauralWSJ0Mix folder and run train.py again, it will download automatically and train smoothly.
But I still don't know the 80 stuck you said and pytest check failure. Thanks.

**Comment by ycemsubakan** [comment]
The noisy version seems to work after your last push. (the data generation) 

But the loss is still kind of stuck:
Capture d’écran, le 2022-08-01 à 10 25 49

**Comment by huangzj421** [comment]
> The noisy version seems to work after your last push. (the data generation)
> 
> But the loss is still kind of stuck: Capture d’écran, le 2022-08-01 à 10 25 49

It was weired cause I just trained and the loss goes down well:
image
My command was:
CUDA_VISIBLE_DEVICES=3,5,6,7 python -m torch.distributed.launch --nproc_per_node=4 train.py hparams/convtasnet-parallel.yaml --data_folder /mnt/test/ --wsj_root /mnt/wsj/ --distributed_launch --distributed_backend='nccl'
Your loss was too big, did you use the dynamic mixing?

**Comment by ycemsubakan** [comment]
> > The noisy version seems to work after your last push. (the data generation)
> > But the loss is still kind of stuck: Capture d’écran, le 2022-08-01 à 10 25 49
> 
> It was weired cause I just trained and the loss went down well: image My command was: CUDA_VISIBLE_DEVICES=3,5,6,7 python -m torch.distributed.launch --nproc_per_node=4 train.py hparams/convtasnet-parallel.yaml --data_folder /mnt/test/ --wsj_root /mnt/wsj/ --distributed_launch --distributed_backend='nccl' Your loss was too big, did you use the dynamic mixing?

No, I am not using dynamic mixing. I am training on a single gpu. Can you try training on a single gpu?

**Comment by huangzj421** [comment]
> > > The noisy version seems to work after your last push. (the data generation)
> > > But the loss is still kind of stuck: Capture d’écran, le 2022-08-01 à 10 25 49
> > 
> > 
> > It was weired cause I just trained and the loss went down well: image My command was: CUDA_VISIBLE_DEVICES=3,5,6,7 python -m torch.distributed.launch --nproc_per_node=4 train.py hparams/convtasnet-parallel.yaml --data_folder /mnt/test/ --wsj_root /mnt/wsj/ --distributed_launch --distributed_backend='nccl' Your loss was too big, did you use the dynamic mixing?
> 
> No, I am not using dynamic mixing. I am training on a single gpu. Can you try training on a single gpu?

It was not so big as well:
image

**Comment by ycemsubakan** [comment]
> 



> > > > The noisy version seems to work after your last push. (the data generation)
> > > > But the loss is still kind of stuck: Capture d’écran, le 2022-08-01 à 10 25 49
> > > 
> > > 
> > > It was weired cause I just trained and the loss went down well: image My command was: CUDA_VISIBLE_DEVICES=3,5,6,7 python -m torch.distributed.launch --nproc_per_node=4 train.py hparams/convtasnet-parallel.yaml --data_folder /mnt/test/ --wsj_root /mnt/wsj/ --distributed_launch --distributed_backend='nccl' Your loss was too big, did you use the dynamic mixing?
> > 
> > 
> > No, I am not using dynamic mixing. I am training on a single gpu. Can you try training on a single gpu?
> 
> It was not so big as well: image

Hm.. weird.. I am not sure why this is happening when I try. This doesn't happen with other recipes. I'll try to understand what's happening.

**Comment by ycemsubakan** [comment]
> > > 
> > 
> > 
> > > > > > The noisy version seems to work after your last push. (the data generation)
> > > > > > But the loss is still kind of stuck: Capture d’écran, le 2022-08-01 à 10 25 49
> > > > > 
> > > > > 
> > > > > It was weired cause I just trained and the loss went down well: image My command was: CUDA_VISIBLE_DEVICES=3,5,6,7 python -m torch.distributed.launch --nproc_per_node=4 train.py hparams/convtasnet-parallel.yaml --data_folder /mnt/test/ --wsj_root /mnt/wsj/ --distributed_launch --distributed_backend='nccl' Your loss was too big, did you use the dynamic mixing?
> > > > 
> > > > 
> > > > No, I am not using dynamic mixing. I am training on a single gpu. Can you try training on a single gpu?
> > > 
> > > 
> > > It was not so big as well: image
> > 
> > 
> > Hm.. weird.. I am not sure why this is happening when I try. This doesn't happen with other recipes. I'll try to understand what's happening.
> 
> Maybe you could remove all the dataset and train it again?

Yes, I already tried that. I generated the dataset from scratch.

**Comment by huangzj421** [comment]
Hi @ycemsubakan, do you have any updates?

**Comment by ycemsubakan** [comment]
@huangzj421 No, I will try to get to it this week.

**Comment by huangzj421** [comment]
> @huangzj421 No, I will try to get to it this week.

Okay, thanks!

**Comment by ycemsubakan** [comment]
With `convtasnet-parallel-noise.yaml`, I observed the following error. Do you see it also? 

image

**Comment by huangzj421** [comment]
> With `convtasnet-parallel-noise.yaml`, I observed the following error. Do you see it also?
> 
> image

No, I guess you should update the automatic downloaded data generation scripts from https://github.com/huangzj421/Binaural-WSJ0Mix or you can remove the old version in your data_folder.

**Comment by ycemsubakan** [comment]
> > With `convtasnet-parallel-noise.yaml`, I observed the following error. Do you see it also?
> > image
> 
> No, I guess you should update the automatic downloaded data generation scripts from https://github.com/huangzj421/Binaural-WSJ0Mix or you can remove the old version in your data_folder.

I see.. It was working earlier. But, this could be an issue for people who use this recipe in the future. Can you automatically check if the folders are correct, and then if not, you can either automatically fix, or at least you need to give an error message so that people know that they should delete the folder, and create a new one.

File: recipes/BinauralWSJ0Mix/separation/train.py
Comment (huangzj421) [inline recipes/BinauralWSJ0Mix/separation/train.py:709]: image
Hi @ycemsubakan, I don't know how to check the scripts. The simliest way is to delete this 'if' command. But I think the dataset generation scripts won't change once this PR merged so we don't need to correct it.

**Comment by ycemsubakan** [comment]
@huangzj421 , all of the .yaml files for me now work. For some reason when I tried with my conda environment, as I mentioned the training was not going down. But with a standard virtual environment it works fine. 

I think you can do the version check for the dataset in a separate PR, and if everything is fine with @mravanelli, we can merge.

**Comment by huangzj421** [comment]
> @huangzj421 , all of the .yaml files for me now work. For some reason when I tried with my conda environment, as I mentioned the training was not going down. But with a standard virtual environment it works fine.
> 
> I think you can do the version check for the dataset in a separate PR, and if everything is fine with @mravanelli, we can merge.

That's great, thanks! 'Separate PR' means opening a new PR?

**Comment by ycemsubakan** [comment]
> > @huangzj421 , all of the .yaml files for me now work. For some reason when I tried with my conda environment, as I mentioned the training was not going down. But with a standard virtual environment it works fine.
> > I think you can do the version check for the dataset in a separate PR, and if everything is fine with @mravanelli, we can merge.
> 
> That's great, thanks! 'Separate PR' means opening a new PR?

Yes. If you already have a solution for dataset versioning, feel free to push it here. If you think it's going to take time, we can open a new PR, after we merge this one.

**Comment by huangzj421** [comment]
> > > @huangzj421 , all of the .yaml files for me now work. For some reason when I tried with my conda environment, as I mentioned the training was not going down. But with a standard virtual environment it works fine.
> > > I think you can do the version check for the dataset in a separate PR, and if everything is fine with @mravanelli, we can merge.
> > 
> > 
> > That's great, thanks! 'Separate PR' means opening a new PR?
> 
> Yes. If you already have a solution for dataset versioning, feel free to push it here. If you think it's going to take time, we can open a new PR, after we merge this one.

Okay, got it. But some checks of PR failed, how can we merge it?

**Comment by ycemsubakan** [comment]
> > > > @huangzj421 , all of the .yaml files for me now work. For some reason when I tried with my conda environment, as I mentioned the training was not going down. But with a standard virtual environment it works fine.
> > > > I think you can do the version check for the dataset in a separate PR, and if everything is fine with @mravanelli, we can merge.
> > > 
> > > 
> > > That's great, thanks! 'Separate PR' means opening a new PR?
> > 
> > 
> > Yes. If you already have a solution for dataset versioning, feel free to push it here. If you think it's going to take time, we can open a new PR, after we merge this one.
> 
> Okay, got it. But some checks of PR failed, how can we merge it?

The checks are failing because of a general problem. It will be fixed soon.

**Comment by huangzj421** [comment]
> > > > > @huangzj421 , all of the .yaml files for me now work. For some reason when I tried with my conda environment, as I mentioned the training was not going down. But with a standard virtual environment it works fine.
> > > > > I think you can do the version check for the dataset in a separate PR, and if everything is fine with @mravanelli, we can merge.
> > > > 
> > > > 
> > > > That's great, thanks! 'Separate PR' means opening a new PR?
> > > 
> > > 
> > > Yes. If you already have a solution for dataset versioning, feel free to push it here. If you think it's going to take time, we can open a new PR, after we merge this one.
> > 
> > 
> > Okay, got it. But some checks of PR failed, how can we merge it?
> 
> The checks are failing because of a general problem. It will be fixed soon.

Okay, thank you!

**Comment by huangzj421** [comment]
Hi @mravanelli , do you have time to check this PR? Cem has reviewed it and everything is fine. Thanks.

**Comment by ycemsubakan** [comment]
Everything seems fine to me. I ran the code one other time. I did some updates to README. I will merge once the tests pass.

**Review by ycemsubakan — APPROVED** [review]

**Pull request merged**
